### PR TITLE
1st_assignment

### PR DESCRIPTION
**Pull request by jeanneomarie**

### CODE REVIEW

**Comment by jeanneomarie** [comment]
i had some questions while doing the hw
i wrote "help" next to them :)
-> for the one with the Fibonnaci numbers I'm not sure of the complexity of my solution, and also I didn't know how large the fibNumbers array should be
-> while for the one with the SumOfMultiples i couldn't say when the overlflow would appear

File: Students/Onofrei I. Ioana Maria/FibonnaciEvenSum/src/fibonnaciSum.java [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+		
+		/**
+		 * same problem, different approach
+		 * @param fibNumbers is an array of integers, in which will store the Fibonnaci numbers
+		 * @return evenSum which is the answer
+		 */
+		
+		//twist
+		//idk if this is in O(n), but still
+		int []fibNumbers = new int[10000000]; //HELP how large the array should be?
```
Comment (AlexBondor) [inline Students/Onofrei I. Ioana Maria/FibonnaciEvenSum/src/fibonnaciSum.java]: You could have placed a breakpoint inside the if on line 47 and see what the value of `i` is at that place. There are 32 numbers < 4mil in fibonacci sequence.. so you could have used 33 as the array size or you could have though about some other stopping condition for the `for` loop on line 44.

File: Students/Onofrei I. Ioana Maria/FibonnaciEvenSum/src/fibonnaciSum.java [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+		
+		
+		/**
+		 * same problem, different approach
+		 * @param fibNumbers is an array of integers, in which will store the Fibonnaci numbers
+		 * @return evenSum which is the answer
+		 */
+		
+		//twist
+		//idk if this is in O(n), but still
```
Comment (AlexBondor) [inline Students/Onofrei I. Ioana Maria/FibonnaciEvenSum/src/fibonnaciSum.java]: `O(n)` means that you're _touching_ all the elements only once. More specific you're computing every single element of your fibonnaci sequence.

File: Students/Onofrei I. Ioana Maria/LargestPalindrome/src/Palindrome.java [hunk 3]
```diff
@@ -0,0 +1,49 @@
+
+public class Palindrome {
+
+	/**
+	 * check if a number is a palindrom or not
+	 * @param number is the one that we are checking
+	 * @param reversedNumber is the one that we form by reversing the original number
```
Comment (AlexBondor) [inline Students/Onofrei I. Ioana Maria/LargestPalindrome/src/Palindrome.java]: this means that your method `isPalindrom` should have a parameter named `reversedNumber` but it doesn't

File: Students/Onofrei I. Ioana Maria/LargestPalindrome/src/Palindrome.java [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+		//I'm using a break with a label in order to make my program get out of the both for loops
+		//i,j are 99, 999,9999 one by one according to the twist that we are doing
+		find_palindrom:
+			for(int i=9999;i>=1000;i--)
+			{
+				for(int j=9999;j>=1000;j--) {
+					number=i*j;
+					if(isPalindrom(number)){
+						System.out.println(number);
+						break find_palindrom;
```
Comment (AlexBondor) [inline Students/Onofrei I. Ioana Maria/LargestPalindrome/src/Palindrome.java]: ugly `break` and label.. not used in development. Maybe extract this code to a method and replace the `break` with a `return`

File: Students/Onofrei I. Ioana Maria/LargestPalindrome/src/Palindrome.java [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+
+		int number;
+
+		
+		//I'm using a break with a label in order to make my program get out of the both for loops
+		//i,j are 99, 999,9999 one by one according to the twist that we are doing
+		find_palindrom:
+			for(int i=9999;i>=1000;i--)
+			{
+				for(int j=9999;j>=1000;j--) {
```
Comment (AlexBondor) [inline Students/Onofrei I. Ioana Maria/LargestPalindrome/src/Palindrome.java]: your solution might be wrong..
check this: [issue explained](https://stackoverflow.com/questions/21286890/find-the-largest-palindrome-made-from-the-product-of-two-3-digit-numbers-javas)

File: Students/Onofrei I. Ioana Maria/SumOfMultiples/src/sumMultiples.java [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+		int limit= limitNumber.nextInt();
+		System.out.println(sum(limit));
+		limitNumber.close();
+
+		/*
+        twist2 - HELP
+        The overflow for an integer appears at 9,568,031
+        But I discovered this only after I compiled my code a few (actually more than a few) times
+        This is why I don't know the answer for long
+		 */
```
Comment (AlexBondor) [inline Students/Onofrei I. Ioana Maria/SumOfMultiples/src/sumMultiples.java]: There is `Integer.MAX_VALUE` and `Long.MAX_VALUE`. If you try to add 1 to any of those values an overflow will occur. You can run a for until your sum reaches any of the values and see for which value the overflow would occur.

File: Students/Onofrei I. Ioana Maria/FibonnaciEvenSum/src/fibonnaciSum.java [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+			}
+			if(fibNumbers[i] % 2 == 0) {
+				evenSum+=fibNumbers[i];
+			}
+		}
+		System.out.println(evenSum);
+		
+		//twist2 
+		//HEEELP
+		// O(log n)??
```
Comment (AlexBondor) [inline Students/Onofrei I. Ioana Maria/FibonnaciEvenSum/src/fibonnaciSum.java]: There is a nice formula for computing even terms of the fibonacci sequence
```
f[i] = 4 * f[i-1] + f[i-2]
f[0] = 0
f[1] = 2
```

**Pull request merged**